/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

syntax = "proto3";

package resdb.utxo;

message UTXOIn {
  int64 prev_id = 1;
  int32 out_idx = 2;
}

message UTXOOut{
  string address = 1;
  int64 value = 2;
  bool spent = 3;
  string pub_key = 4;
}

message UTXO {
  repeated UTXOIn in = 1;
  repeated UTXOOut out = 2;
  string sig = 3; // signed by the owner.
  string address = 4;
  int64 transaction_id = 5;
}
